### PR TITLE
SD tutorial and benchmarks

### PR DESCRIPTION
**Pull request by jeffra**
* SD benchmark scripts
* SD tutorial showing how to deploy and evaluate a deployment
* Add MII package version that can be checked at runtime
* Fix GPRC max message size issue that I ran into when running multi-batch SD
* Added optional mii config to skip model validation checks (this currently takes >15sec and is not always needed)

### CODE REVIEW

File: examples/benchmark/txt2img/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
+# Generate Stable Diffusion Images with < 1 Second per Image
+
+<div align="center">
+ <img src="../../../docs/images/sd-hero.png#gh-light-mode-only">
+ <img src="../../../docs/images/sd-hero-dark.png#gh-dark-mode-only">
+</div>
+
+In this tutorial you will learn how to deploy [Stable Diffusion](https://huggingface.co/CompVis/stable-diffusion-v1-4) with state-of-the-art performance optimizations from DeepSpeed. Specifically, you will use both [DeepSpeed Inference](https://github.com/microsoft/deepspeed) and [DeepSpeed-MII](https://github.com/microsoft/deepspeed-mii) to your deployment. In addition to deploying we will perform several performance evaluations.
```
Comment (mrwyattii) [inline examples/benchmark/txt2img/README.md:8]: The second sentence here has a mistake: "Specifically, you will use both DeepSpeed Inference and DeepSpeed-MII to your deployment". Perhaps you meant "... to create your deployment"?

File: examples/benchmark/txt2img/README.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+This tutorial and related results are using Azure [ND96amsr\_A100\_v4](https://learn.microsoft.com/en-us/azure/virtual-machines/nda100-v4-series) instances with NVIDIA A100-80GB GPUs. We observe similar performance on [ND96asr\_v4](https://learn.microsoft.com/en-us/azure/virtual-machines/nda100-v4-series) instances with NVIDIA A100-40GB GPUs. In addition all of the techniques described here have also been successfully deployed on NVIDIA RTX A6000 GPUs as well.
+
+## Outline
+* [Environment and dependency setup](#environment-setup)
```
Comment (mrwyattii) [inline examples/benchmark/txt2img/README.md:13]:
```suggestion
* [Environment and dependency setup](#environment-and-dependency-setup)
```

File: examples/benchmark/txt2img/README.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+```python
+import mii
+generator = mii.mii_query_handle("sd_deploy")
+prompt = "a photo of an astronaut riding a horse on mars"
+image = generator.query({'query': prompt}).images[0]
+image.save("horse-on-mars.png")
+```
+
+We've packaged up all that you need to deploy, query, and tear down an SD MII deployment for you in [mii-sd.py](mii-sd.py) which we will refer to going forward. You can run this example via:
```
Comment (mrwyattii) [inline examples/benchmark/txt2img/README.md:143]:
```suggestion
We've packaged up all that you need to deploy, query, and tear down an SD MII deployment in [mii-sd.py](mii-sd.py) which we will refer to going forward. You can run this example via:
```

File: mii/__init__.py [hunk 4]
```diff
@@ -7,3 +7,9 @@
 
 from .config import MIIConfig
 from .grpc_related.proto import modelresponse_pb2_grpc
+
+__version__ = "0.0.0"
+try:
+    from .version import __version__
+except ImportError:
+    pass
```
Comment (mrwyattii) [inline mii/__init__.py:15]: What is the purpose of this?

**Review by mrwyattii — APPROVED** [review]
LGTM, just a few minor suggestions to tutorial

File: mii/__init__.py (hunk 4 above)
Comment (Thomas-MMJ) [reply mii/__init__.py:15]: elsewhere the author does

`assert version.parse(mii.__version__) >= version.parse("0.0.3"), "mii must be 0.0.3+"

so this provides a fake version number for versions prior to 0.0.3 that don't have one to allow the assert to work

**Comment by Suhail** [comment]
I'd recommend you mention how many steps are taken for the benchmark comparison. I didn't immediately find it so apologies if I missed it too.

**Comment by jeffra** [comment]
Thank you @Suhail for the feedback, I’ll update the tutorial before merging to add this info.

One thing for anyone reading this PR, we have pending changes coming later today on the DeepSpeed side related further performance updates and bug fixes. We’ll merge this PR once those our live.

**Pull request merged**